### PR TITLE
Run unit and E2E tests on MacOS

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it

This pull request adds GitHub workflows to run unit and E2E tests on macOS.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
The new GitHub workflow run as check for the unit tests  on macOS.

### Release note

```release-note

```

### CODE REVIEW

File: .github/workflows/cli_core_unit_test_mac.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+on:
+  pull_request:
+    branches: [main, release-*]
+  push:
+    branches: [ main, test*, release-*]
+
+jobs:
+  build:
+    name: Tanzu CLI Unit Tests
+    runs-on: macos-latest
```
Comment (anujc25) [inline .github/workflows/cli_core_unit_test_mac.yaml:12]: You should not required to create a new unit test job to run the same thing on a different OS. You can use the `strategy` field to specify this. https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-multi-dimension-matrix

You should be able to do something like the below in the `cli_core_unit_test.yaml` file to run the pipeline on both OS.

```suggestion
    strategy:
      matrix:
        os: [macos-latest, ubuntu-latest]
    runs-on: ${{ matrix.os }}
```

**Review by anujc25 — CHANGES_REQUESTED** [review]

File: .github/workflows/cli_core_unit_test_mac.yaml (hunk 1 above)
Comment (chandrareddyp) [reply .github/workflows/cli_core_unit_test_mac.yaml]: thanks for suggestion, fixed.

File: .github/workflows/cli_core_e2e_test.yaml [hunk 2]
```diff
@@ -8,8 +8,11 @@ on:
 
 jobs:
   build:
-    name: Tanzu CLI Core E2E Tests
+    name: Tanzu CLI Core E2E Tests on Mac and Ubuntu
```
Comment (anujc25) [inline .github/workflows/cli_core_e2e_test.yaml:11]:
```suggestion
    name: Tanzu CLI Core E2E Tests
```

**Review by anujc25 — APPROVED** [review]

**Review by marckhouzam — COMMENTED** [review]
Is it the unit test or the e2e tests we want to run on macOS?  The PR title and description says unit tests but the code change is for E2E

**Comment by chandrareddyp** [comment]
> Is it the unit tests or the e2e tests we want to run on macOS? The PR title and description says unit tests but the code change is for E2E

Updated, this PR to execute unit and e2e tests in MacOS.

**Review by marckhouzam — APPROVED** [review]
LGTM
Thanks @chandrareddyp

**Pull request merged**